### ISSUE TITLE
Sync Chain has zero protection (i.e. no password)

### ISSUE THREAD

**Issue by LuKks** [brave/brave-browser#29544]
## Description 
When you `Start a new Sync Chain`, it gives you 25 words. Let's say you share the computer for a moment, someone in very few seconds could just press to view the `Sync Code` again, and later have full access on the future to all your history, bookmarks, etc without you knowing until it's already late.

It's kind of a similar when someone can grab an unlocked phone and log in into WhatsApp Web without the other person knowing it right away (of course there is a notification, etc), but the idea is that even if someone doesn't take perfect precautions then it should still be secure on Brave.

## Steps to Reproduce
It's not a bug to reproduce, it's just how it currently works the `View Sync Code` button.

## Actual result:
![image](https://user-images.githubusercontent.com/12686176/230231733-5a1a78e5-9bcb-46bc-bac6-6943212c86f4.png)

## Expected result:
I would expect an optional preference with two or three options for the `Sync Code` behaviour like:
- To be shown only once
- To be protected by password
- Without protection

## Desktop Brave version:
```
Brave | 1.50.114 Chromium: 112.0.5615.49 (Official Build) (64-bit)
Revision | bd2a7bcb881c11e8cfe3078709382934e3916914-refs/branch-heads/5615@{#936}
OS | Linux
```

## Version/Channel Information:
Release channel

**Comment by LuKks** [comment]
A single user like me doesn't matter, but this is the only thing that is stopping me to switch to Brave! Would love to see some improvement around Sync

**Comment by pramirez92** [comment]
This needs to be addressed. It's a huge flaw in security.

For Windows, I think the obvious solution is to require the user to enter their Windows login password before the Sync Code appears. This is the current behavior before you are allowed to view individual saved passwords in the Brave Password Manager. Why not implement exactly the same thing for the Sync Code?

**Comment by ddanaj** [comment]
I was just switching to Brave but this stopped me and I am going to explore other options.

Showing Sync Code without any protection is a big flaw. If anyone gets to my PC or mobile phone just for few seconds, they can get access to literally everything, including passwords, very easily. Device protection is obviously important, but things can happen (you let someone check a video from holiday, go grab a water and all passwords are compromised).

Sync Code should be hidden behind additional protection. For example in the same way as passwords - you need to "login" again on Windows or "unblock" phone on Android to see a password.

**Comment by Saoiray** [comment]
@rebron I wanted to circle back to this. If I can simplify, the browser asks for OS password when we try to Export. Is there a way to use that same mechanism when accessing Sync? 

At least hoping can advise if would be fairly simple lift or heavy. And if anyone might help understand priority. But definitely is something I see requested on occasion. Recent example is https://community.brave.com/t/sync-security-question/535995

**Comment by AguiarPhys** [comment]
I've just noticed this flaw as well when adding a new device. It's way too easy to steal someone's entire password manager log. It needs at least an extra security step + push notifications to all devices.

**Comment by 7eventech77** [comment]
Agree to this as well, honestly surprised why this hasn't been implemented. Creates a huge barrier for businesses to begin to use Brave since there is no way they are going to allow people to sync passwords and bookmarks, etc. without some sort of security if they want to use this type feature.

**Comment by bsclifton** [comment]
cc: @fmarier

**Comment by fmarier** [comment]
The way we are planning to fix this is to treat Sync codes the same as passwords in the password manager. That is, on most platforms, you are prompted for a password/biometrics before viewing passwords. This means that if on a given platform you are asked to authenticate before seeing passwords, then you should also be asked to authenticate before viewing the Sync code (if password sync'ing is enabled).

At the moment, that's the behavior on iOS. On Android, https://github.com/brave/brave-browser/issues/30268 started the work but it's incomplete since it's only enabling password syncing that's behind auth. Once enabled, you can view the Sync code without auth.

On desktop, there is no prompt on Windows or Mac even though these platforms do prompt inside of the password manager.

**Comment by fmarier** [comment]
I'm going to close this issue as a "duplicate" because we are tracking the Android and desktop cases into two separate issues. If there are any additional problems you see, then please comment (or alternatively, open a new issue).
